event_session/models/event_session.py

```diff
@@ -0,0 +1,275 @@
+
+    @api.multi
+    def name_get(self):
+        """Redefine the name_get method to show the event name with the event
+        session.
+        """
+        res = []
+        for item in self:
+            res.append((item.id, "[%s] %s" % (item.event_id.name, item.name)))
+        return res
+
+    @api.model
+    def create(self, vals):
+        if not vals.get('event_mail_ids', False):
+            vals.update({
+                'event_mail_ids':
+                    self._session_mails_from_template(vals['event_id'])
+            })
+        return super(EventSession, self).create(vals)
+
+    @api.multi
+    @api.depends('seats_max', 'registration_ids.state')
+    def _compute_seats(self):
+        """Determine reserved, available, reserved but unconfirmed and used
+        seats by session.
+        """
+        # initialize fields to 0
+        for session in self:
+            session.seats_unconfirmed = session.seats_reserved = \
+                session.seats_used = session.seats_available = 0
+        # aggregate registrations by event session and by state
+        if self.ids:
+            state_field = {
+                'draft': 'seats_unconfirmed',
+                'open': 'seats_reserved',
+                'done': 'seats_used',
+            }
+            query = """
+                SELECT session_id, state, count(session_id)
+                FROM event_registration
+                WHERE session_id IN %s AND state IN ('draft', 'open', 'done')
+                GROUP BY session_id, state """
+            self._cr.execute(query, (tuple(self.ids),))
+            for session_id, state, num in self._cr.fetchall():
+                session = self.browse(session_id)
+                session[state_field[state]] += num
+        # compute seats_available
+        for session in self:
+            if session.seats_max > 0:
+                session.seats_available = session.seats_max - (
+                    session.seats_reserved + session.seats_used)
+            session.seats_expected = (
+                session.seats_unconfirmed + session.seats_reserved +
+                session.seats_used)
+            session.seats_available_expected = (
+                session.seats_max - session.seats_expected)
+            session.seats_available_pc = (
+                session.seats_expected * 100 / float(session.seats_max))
+
+    @api.multi
+    @api.depends('date_tz', 'date_begin')
+    def _compute_date_begin_located(self):
+        for session in self:
+            if session.date_begin:
+                self_in_tz = session.with_context(
+                    tz=(session.date_tz or 'UTC')
+                )
+                date_begin = fields.Datetime.from_string(session.date_begin)
+                session.date_begin_located = fields.Datetime.to_string(
+                    fields.Datetime.context_timestamp(self_in_tz, date_begin),
+                )
+            else:
+                session.date_begin_located = False
```

Comment:
No need to assign default value, so instead just do `for session in self.filtered('date_begin'):`